### PR TITLE
Smart recording rebase

### PR DESCRIPTION
**Pull request by EddyGharbi**

### CODE REVIEW

File: obs-studio-client/source/sceneitem.cpp [hunk 1]
```diff
@@ -265,6 +267,136 @@ Nan::NAN_METHOD_RETURN_TYPE osn::SceneItem::SetSelected(Nan::NAN_METHOD_ARGS_TYP
...
+		return;
+
+	std::vector<ipc::value> response =
+	    conn->call_synchronous_helper("SceneItem", "IsStreamVisible", std::vector<ipc::value>{ipc::value(item->itemId)});
+
+	if (!ValidateResponse(response))
+		return;
+	bool streamVisible = !!response[1].value_union.ui32;
+
+	sid->streamVisibleChanged = false;
```
Comment (pdamasceno) [inline obs-studio-client/source/sceneitem.cpp]: Best to verify if `sid` is valid before changing its values

**Review by summeroff — APPROVED** [review]

File: obs-studio-client/source/sceneitem.cpp [hunk 2]
```diff
@@ -265,6 +267,136 @@ Nan::NAN_METHOD_RETURN_TYPE osn::SceneItem::SetSelected(Nan::NAN_METHOD_ARGS_TYP
...
+		return;
+
+	std::vector<ipc::value> response = conn->call_synchronous_helper(
+	    "SceneItem", "IsRecordingVisible", std::vector<ipc::value>{ipc::value(item->itemId)});
+
+	if (!ValidateResponse(response))
+		return;
+	bool recordingVisible = !!response[1].value_union.ui32;
+
+	sid->recordingVisibleChanged = false;
```
Comment (pdamasceno) [inline obs-studio-client/source/sceneitem.cpp:363]: Same thing here, verify if `sid` is valid before changing its values

File: obs-studio-client/source/sceneitem.cpp (hunk 1 above)
Comment (EddyGharbi) [reply obs-studio-client/source/sceneitem.cpp]: Good catch! Fixed 👍

File: obs-studio-client/source/sceneitem.cpp (hunk 2 above)
Comment (EddyGharbi) [reply obs-studio-client/source/sceneitem.cpp:363]: Good catch! Fixed 👍

**Review by pdamasceno — APPROVED** [review]

**Pull request merged**